src/document/utils/validation.ts

```diff
@@ -0,0 +1,26 @@
+import { DocumentOperations, Action, ValidationError } from '../types';
+
+export function validateOperations<A extends Action>(
+    operations: DocumentOperations<A>,
+) {
+    const errors: ValidationError[] = [];
+
+    const ops = [...operations.global, ...operations.local].sort(
+        (a, b) => a.index - b.index,
```

Comment:
We should do this per scope. Global/local operations have their own index.

Reply:
forgot that change, here is the fix: [4084461](https://github.com/powerhouse-inc/document-model/pull/34/commits/4084461b3b1465c5152ea6bdb00ea4781330e13a)